kubernetes/elastic-helm/configmap-elastic.yaml

```diff
@@ -21,6 +21,21 @@ data:
     extensions:
     processors:
       batch: {}
+      resource:
+        attributes:
+          - key: deployment.environment
+            value: "opentelemetry-demo"
+            action: upsert
+      attributes:
+        actions:
+          - key: http.url
+            action: extract
+            pattern: '^(?P<short_url>https?://[^/]+(?:/[^/]+)*)(?:/(?P<url_truncated_path>[^/?]+/[^/?]+))(?:\?|/?$)'
+      transform:
+        trace_statements:
+          - context: span
+            statements:
+              - set(name, attributes["url_truncated_path"])
```

Comment:
Could you provide more context about this change? I think the microservices are setting their corresponding span names, and they should follow the format {method} {format} as defined in https://opentelemetry.io/docs/specs/semconv/http/http-spans/#name

Reply:
@rogercoll actually this is a good point, I was aiming for the guidance in here https://github.com/open-telemetry/opentelemetry-specification/blob/v1.33.0/specification/trace/api.md#span see

get_account | Good, and account_id=42 would make a nice Span attribute
get_account/{accountId} | Also good (using the "HTTP route")

I'll be honest I don't really understand the guidance in the semconv as it seems to contradict this. Honestly though at the moment in elastic the top transaction name becomes "HTTP GET" which is incredibly general and not useful for monitoring imo. 

What do you think?

Reply:
To illustrate. This piece of configuration makes the transaction names more specific from this

![image (66)](https://github.com/user-attachments/assets/ac6a5dde-fd37-4439-b256-93215c2e61f2)

To this:

![image (67)](https://github.com/user-attachments/assets/ab8a38df-3f87-4b3b-8eb0-6eb9796d827c)

imo much more useful. I don't know if the problem is the way the demo is built or the way Elastic interprets the data?

Reply:
Ah, I see. That's a really good point, thanks for raising that. I think the issue is in how the upstream services generate the root spans and which name do they set. The issue seems to only be happening for HTTP requests, GRPC services set a more human-readable name. I have just raised this issue in the upstream repository: https://github.com/open-telemetry/opentelemetry-demo/issues/1676 

Wdyt if we move this discussion upstream? Using the `transform` processor in the collector might patch the issue, but I think it would be better to fix it in the root and have a concise name across all generated spans (HTTP and GRPC).

Reply:
Thanks for doing that @rogercoll makes perfect sense.